src/util/circular_buffer.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <sys/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// forward declaration
+typedef struct _CircularBuffer _CircularBuffer;
+typedef _CircularBuffer* CircularBuffer;
+
+// Creates a new circular buffer, with `cap` items of size `item_size`
+CircularBuffer CircularBuffer_New
```

Comment:
All these function signatures seem to deviate from our code convention.
Consider aligning them.

Reply:
Not true. See IndexSpec.c, references.c etc.
Once we will have a formatter we will be more aligned on this. What format are you referring to?